lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -209,21 +213,24 @@ def debug_message(cause = nil, info_hash = {})
         helpful_info << "  - Events: #{DEBUG_RESOURCE_NOT_FOUND_MESSAGE}"
       end
 
-      if supports_logs?
+      if supports_debug_logs?
         if ENV[DISABLE_FETCHING_LOG_INFO]
           helpful_info << "  - Logs: #{DISABLED_LOG_INFO_MESSAGE}"
-        elsif @logs.blank? || @logs.values.all?(&:blank?)
+        elsif @debug_logs.blank?
           helpful_info << "  - Logs: #{DEBUG_RESOURCE_NOT_FOUND_MESSAGE}"
         else
-          sorted_logs = @logs.sort_by { |_, log_lines| log_lines.length }
-          sorted_logs.each do |identifier, log_lines|
-            if log_lines.empty?
-              helpful_info << "  - Logs from container '#{identifier}': #{DEBUG_RESOURCE_NOT_FOUND_MESSAGE}"
+          container_logs = @debug_logs.container_logs.sort_by { |c| c.lines.length }
```

Comment:
just out of curiosity, why do we want to sort like this?

Reply:
IIRC we added this sorting for cases where containerA dumps 250 lines of useless crap (e.g. services ping logs) and containerB dumps a shorter number of useful lines, which users didn't even notice. The idea is that even if containerB's logs are less useful, you'll at least notice you have both. I don't feel strongly about this.